### ISSUE TITLE
Service injected directly in component is not mocked

### ISSUE THREAD

**Issue by Babeetlebum** [ike18t/ng-mocks#1507]
When providing a service directly in a component 
```typescript
@Component({
  ...
  providers: [HelloService],
})
```
Then the conventional mocking techniques don't work anymore

### `.mock`

I would expect `MockBuilder(HelloComponent).mock(HelloService)` to mock the `HelloService` but test doesn't compile

### `.provide` with `useValue`

I would expect `MockBuilder(HelloComponent).provide({ provide: HelloService, useValue: FakeService })` to mock the `HelloService` but the real service is provided instead

### `MockInstance`

I would expect `MockBuilder(HelloComponent).provide(HelloService)` followed by `MockInstance(HelloService, () => FakeService)` to mock the `HelloService` but the real service is provided instead

Here's a stackblitz to reproduce the issue: https://stackblitz.com/edit/angular-ivy-rgnbt7?file=src%2Fapp%2Fhello%2Fhello.component.spec.ts

**Comment by satanTime** [comment]
Hi,

actually it works on my side as expected:
- `.mock` mocks component's poviders
- `.provide` adds a global provider and doesn't affect component's providers

https://stackblitz.com/edit/github-tt2yvq?file=src/test.spec.ts

the right way is to use `.mock`, because `.provide` doesn't cover component providers (likewise angular modules behave).

**Comment by satanTime** [comment]
Hi @Babeetlebum,

I'll close the issue for now. Feel free to reopen it once you have more info or we should discuss anything.

**Comment by Babeetlebum** [comment]
I still haven't figured out why it works in your environment but not in mine. Also in my real app it still doesn't work with `.mock`. I migrated the project to jest recently and the error persists. I'll keep looking into it.

I just want to mention that the example in your stackblitz fails when using `viewProviders` instead of `providers`, I forked the stackblitz here : https://stackblitz.com/edit/github-tt2yvq-dlr7wc?file=src/test.spec.ts
